### PR TITLE
add capability to limit output size

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

File: test/PrettyPrinting-test.jl [hunk 1]
```diff
@@ -103,47 +122,70 @@
...
+   function limited_latex_string(x)
+      x = AbstractAlgebra.canonicalize(x)
+      b = IOBuffer()
+      c = IOContext(b, :size_limit => 30)
+      AbstractAlgebra.show_obj(c, MIME("text/latex"), x)
+      return String(take!(b))
+   end
+
+   e = Expr(:call, :+, [:(x^$i) for i in 1:200]...)
+   @test length(limited_latex_string(e)) < 200
```
Comment (thofma) [inline test/PrettyPrinting-test.jl:178]: Why not test the actual string?

Comment (tthsqe12) [reply test/PrettyPrinting-test.jl:178]: I don't know what the string is.

**Comment by tthsqe12** [comment]
So I am happy with this. Note that I have avoided touching the generic folder since there is work going on there. Also, there are some explicit string tests, but I see no reason for the explicit tests @thofma mentioned.

File: test/PrettyPrinting-test.jl (hunk 1 above)
Comment (tthsqe12) [reply test/PrettyPrinting-test.jl:187]: I could tell you what the string is after looking at the output of the function, but it is not predictable what it is exactly.

**Comment by tthsqe12** [comment]
I'm done with this.

**Pull request merged**